src/lazy/text/buffer.rs

```diff
@@ -2276,8 +2328,9 @@ mod tests {
             P: Parser<TextBufferView<'data>, O, IonParseError<'data>>,
         {
             let result = self.try_match(parser);
-            let (_remaining, match_length) = result
-                .unwrap_or_else(|_| panic!("Unexpected parse fail for input '{}'", self.input));
+            let (_remaining, match_length) = result.unwrap_or_else(|e| {
```

Comment:
🗺️ This is a developer experience improvement. It causes a test failure to print the whole error, not just "Unexpected parse fail for input".